lib/tree_sitter/mixins/language.rb

```diff
@@ -0,0 +1,161 @@
+      #    #   .vendor/parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   vendor/tree-sitter-parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   vendor/parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   tree-sitter-parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /opt/local/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /opt/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /usr/local/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /usr/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    java = TreeSitter.language('java')
+      #
+      # @example (TreeStand) Load java from a configured path
+      #    # This will look for:
+      #    #
+      #    #   /my/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    TreeStand.config.parser_path = '/my/path'
+      #    java = TreeStand::Parser.language('java')
+      #
+      # @example (TreeStand) Load java from environment variables
+      #    # This will look for:
+      #    #
+      #    #   /my/forced/env/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /my/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    # … and the same works for the default paths if `TreeStand.config.parser_path`
+      #    # was `nil`
+      #    ENV['TREE_SITTER_PARSERS'] = '/my/forced/env/path'
+      #    TreeStand.config.parser_path = '/my/path'
+      #    java = TreeStand::Parser.language('java')
+      #
+      # @param name [String] the name of the parser.
+      #   This name is used to load the symbol from the compiled parser, replacing `-` with `_`.
+      #
+      # @return [TreeSitter:language] a language object to use in your parsers.
+      #
+      # @raise [RuntimeError] if the parser was not found.
+      #
+      # @see search_for_lib
+      def language(name)
+        lib = search_for_lib(name)
+
+        if lib.nil?
+          raise <<~MSG.chomp
+            Failed to load a parser for #{name}.
+
+            #{search_lib_message}
+          MSG
+        end
+
+        # We know that the bindings will accept `lib`, but I don't know how to tell sorbet
+        # the types in ext/tree_sitter where `load` is defined.
+        TreeSitter::Language.load(name.tr('-', '_'), T.unsafe(lib))
+      end
+
+      # The platform-specific extension of the parser.
+      # @return [String] `dylib` or `so` for mac or linux.
+      def ext
+        case Gem::Platform.local.os
+        in   /darwin/ then 'dylib'
+        else               'so'
+        end
+      end
+
+      private
+
+      # The library directories we need to look into.
+      #
+      # @return [Array<Pathname>] the list of candidate places to use when searching for parsers.
+      #
+      # @see ENV_PARSERS
+      # @see LIBDIRS
+      def lib_dirs = [*TreeSitter::ENV_PARSERS, *TreeSitter::LIBDIRS].compact
```

Comment:
That compact is weird.  If there's some `nil` here, it comes from the constants, which should not contain `nil`.

Reply:
Yes you're right.